### DISCUSSION TITLE
Flickable - no animated scrolling after releasing the mouse

### DISCUSSION THREAD

**Discussion by oklemencic** [slint-ui/slint#947] [General] [answered]
Hi everyone, 

I am currently evaluating Slint to see if it fits for our embedded touch-based products. I have tried a few things and the Slint language is generally really great :)

One thing which did not work for me is the animated scrolling after the mouse is released. 

I have looked at the sources and it seems there should be an animation of the viewport:
https://github.com/slint-ui/slint/blob/4230ac2572057198e5fd41b4dea88244ebbc5c83/internal/core/flickable.rs#L158-L161

In my demo the flickable only scrolls a very small distance. I tried using the GL backend and the QT backend (both on Windows). 
Is this a bug?

```
    f := Flickable {
        width:  800px;
        height: 420;
        viewport_width: parent.width;
        viewport_height: 2100px;
        interactive: true;

        for data[index] in [
            { text: "Blue", color: #0000ff},
            { text: "Red", color: #ff0000},
            { text: "Green", color: #00ff00},
            { text: "Yellow", color: #ffff00 },
            { text: "Blue", color: #0000ff},
            { text: "Red", color: #ff0000},
            { text: "Green", color: #00ff00},
            { text: "Yellow", color: #ffff00 },
            { text: "Yellow", color: #ffff00 },
            { text: "Blue", color: #0000ff},
            { text: "Red", color: #ff0000},
            { text: "Green", color: #00ff00},
            { text: "Yellow", color: #ffff00 },
        ] : TouchArea {
            y: 10px + index * 200px;
            x: 10px;
            width: 100px;
            height: 90px;
            Rectangle {
                background: parent.pressed ? blue : parent.has_hover ? green : data.color;
            }
        }
    }
```

**Comment by ogoffart** [comment, marked as answer]
It seems it is indeed a bug. The computation of the speed might be wrong and the animation might be too short or a combination of the two.

**Reply by oklemencic** [reply]
Thanks. Tested the new version with the fix https://github.com/slint-ui/slint/commit/efd6c63b8e899710aa2de9d886119c4c0283f9d7 and this works fine :)

**Comment by oklemencic** [comment]
Thanks for the quick reply.

I tried to adapt the flickable source code at https://github.com/slint-ui/slint/blob/4230ac2572057198e5fd41b4dea88244ebbc5c83/internal/core/flickable.rs#L153-L169 and replaced everything with this code, but unfortunately I get the same behavior. So I guess it's something "deeper". 

```
let final_y = if dist.y > 0.0 { 0. } else { -1000.0 };
println!("Flickable: mouse released. Pos: {} Dist: {}, Animation final value: {}", event.pos().unwrap().y, dist.y, final_y);
let anim = PropertyAnimation {
                duration: 2000,
                easing: EasingCurve::Linear,
                ..PropertyAnimation::default()
            };
(Flickable::FIELD_OFFSETS.viewport + Rectangle::FIELD_OFFSETS.y)
                .apply_pin(flick)
                .set_animated_value(final_y, anim);
```

I now also tried to manually set the viewport-y upon button click and added an animate directive like shown below, but this only generated only ``self->f_6.viewport.y.set((- 1000))`` and does not generate something with ``set_animated_value``:

```
Button {
            text: "set -1000px";
            width: 100px;
            height: 40px;
            clicked => { 
                debug("set -1000px");
                f.viewport-y = -1000px;
             }
        }

f := Flickable {
...
animate viewport.y {
            duration: 500ms;
        }

}
```

Is there some "configuration" which properties can be animated? 
Shall I create a issue for this?

**Reply by ogoffart** [reply]
Maybe the bug is somewhere else. I'll have to investigate. I filled issue #951
